File: app/src/main/java/com/innovation4you/napking/ui/adapter/BaseViewHolder.java
package com.innovation4you.napking.ui.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import butterknife.ButterKnife;

public class BaseViewHolder extends RecyclerView.ViewHolder {

	public BaseViewHolder(final View itemView, final View.OnClickListener onClickListener) {
		super(itemView);
		ButterKnife.bind(this, itemView);
		itemView.setOnClickListener(onClickListener);
	}
}

File: app/src/main/java/com/innovation4you/napking/ui/adapter/BaseViewHolder.kt
package com.innovation4you.napking.ui.adapter

import android.support.v7.widget.RecyclerView
import android.view.View

import butterknife.ButterKnife

open class BaseViewHolder(itemView: View, onClickListener: View.OnClickListener) : RecyclerView.ViewHolder(itemView) {

    init {
        ButterKnife.bind(this, itemView)
        itemView.setOnClickListener(onClickListener)
    }
}